Brokerages/IOrderBookUpdater.cs

```diff
@@ -0,0 +1,60 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Brokerages
+{
+    /// <summary>
+    /// Represents an order book interface for a security.
```

Comment:
This documentation seems to be inconsistent with the  behavior described in the interface. The interface provides the ability to _update_ and to be alerted about updates (via events), but _does not_ provide any read access to the actual order book data.

That being said, is there any particular reason why the bids and asks aren't exposed, for example, we _could_ add two methods, `GetBids` and `GetAsks` as `IEnumerable<KeyValuePair<K, V>>`

Also, it's OK if this abstraction is solely intended for providing updates, we'll just want to update the documentation here to match the defined behavior of the interface.

Reply:
The main concern of this decision was #997. We might want to implement `IOrderBook ` to Lean core and make it more specific (as `SortedDictionary<decimal, decimal>` ) in order to use orderbook within algorithms. Having `GetBids` and `GetAsks` as generic methods is good enough for using them within brokerages and/or brokerage-specific algorithms, but wouldn't like to introduce this limitation and keep all algorithms compatible with all brokerages.